src/core/forces.cpp

```diff
@@ -26,105 +26,68 @@
 
 #include "EspressoSystemInterface.hpp"
 
-#include "p3m_gpu.hpp"
-#include "maggs.hpp"
-#include "forces_inline.hpp"
 #include "electrokinetics.hpp"
+#include "forces_inline.hpp"
+#include "maggs.hpp"
+#include "p3m_gpu.hpp"
 
 #include <cassert>
 ActorList forceActors;
 
-void init_forces()
-{
-  Cell *cell;
-  Particle *p;
-  int np, c, i;
-
-  /* The force initialization depends on the used thermostat and the
-     thermodynamic ensemble */
+void init_forces() {
+/* The force initialization depends on the used thermostat and the
+   thermodynamic ensemble */
 
 #ifdef NPT
   /* reset virial part of instantaneous pressure */
-  if(integ_switch == INTEG_METHOD_NPT_ISO) 
+  if (integ_switch == INTEG_METHOD_NPT_ISO)
     nptiso.p_vir[0] = nptiso.p_vir[1] = nptiso.p_vir[2] = 0.0;
 #endif
 
-
   /* initialize forces with langevin thermostat forces
      or zero depending on the thermostat
      set torque to zero for all and rescale quaternions
   */
-  for (c = 0; c < local_cells.n; c++) {
-    cell = local_cells.cell[c];
-    p  = cell->part;
-    np = cell->n;
-    for (i = 0; i < np; i++)
-      init_local_particle_force(&p[i]);
+  for (auto &p : local_cells.particles()) {
+    init_local_particle_force(&p);
```

Comment:
Yes, but this has requirements on the body that must be checked individually. So you can't make this parallel without touching it anyway.